File: unimonpark/unimonpark-frontend/src/types/vehiculo.ts
export type CategoriaPersona =
    | "ESTUDIANTE"
    | "DOCENTE_ADMINISTRATIVO"
    | "DOCENTE_ADMINISTRATIVO_EXTERNO"
    | "CENTRO_OBRERO"
    | "EXTERNO";

export interface Vehiculo {
    idVehiculo: number;
    idUsuario: number | null;
    nombreUsuario?: string | null;
    idExterno?: number | null;
    nombreExterno?: string | null;
    documentoExterno?: string | null;
    idTipoVehiculo: number;
    nombreTipoVehiculo?: string | null;
    placa: string | null;
    numeroFicha?: string | null;
    marca: string | null;
    modelo: string | null;
    color: string | null;
    categoriaPersona: CategoriaPersona;
    activo: boolean;
    fechaCreacion?: string;
}

export interface VehiculoPayload {
    idUsuario?: number | null;
    idExterno?: number | null;
    idTipoVehiculo: number;
    placa: string | null;
    marca: string;
    modelo: string;
    color: string;
    categoriaPersona: CategoriaPersona;
    activo: boolean;
}

File: unimonpark/src/main/java/co/edu/unimonserrate/unimonpark/entity/Vehiculo.java
package co.edu.unimonserrate.unimonpark.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import co.edu.unimonserrate.unimonpark.enums.CategoriaPersona;

@Entity
@Table(name = "vehiculos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Vehiculo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idVehiculo;

    @Column(unique = true)
    private String placa;

    private String marca;

    private String modelo;

    private String color;

    @Column(nullable = false)
    private Boolean activo;

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;

    @Enumerated(EnumType.STRING)
    @Column(name = "categoria_persona", nullable = false, length = 35)
    private CategoriaPersona categoriaPersona;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_usuario", nullable = false)
    private Usuario usuario;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_tipo_vehiculo", nullable = false)
    private TipoVehiculo tipoVehiculo;
}